apps/map/components/map/index.tsx

```diff
@@ -0,0 +1,194 @@
+"use client"
+import React, { useState, useMemo, useEffect, useCallback, useRef } from "react"
+import { GoogleMap, MarkerF, InfoWindow, useLoadScript } from "@react-google-maps/api"
+import { BusinessMapMarkersQuery } from "@/services/galoy/graphql/generated"
+import Link from "next/link"
+import { theme } from "./map-theme"
+import { env } from "@/env"
+
+type MapComponentProps = {
+  mapData: BusinessMapMarkersQuery["businessMapMarkers"]
+}
+
+export default function MapComponent({ mapData }: MapComponentProps) {
+  const mapRef = useRef<google.maps.Map>()
+  const [selectedMarker, setSelectedMarker] = useState<
+    BusinessMapMarkersQuery["businessMapMarkers"][number] | null
+  >(null)
+  const [currentLocation, setCurrentLocation] = useState({
+    lat: 13.7942,
+    lng: -88.8965,
+  })
+  const libraries = useMemo(() => ["places"], [])
+  const onMapLoad = useCallback((map: google.maps.Map) => {
+    mapRef.current = map
+  }, [])
+
+  const mapOptions = useMemo<google.maps.MapOptions>(
+    () => ({
+      disableDefaultUI: true,
+      clickableIcons: true,
+      zoomControl: true,
+      maxZoom: 38,
+      minZoom: 3,
+      styles: [...theme, ...unwantedMarkers],
```

Comment:
what does `unwantedMarkets` means? why is it useful?

Reply:
Remove unnecessary noise/markers from maps. Otherwise, there is too much information on the map.

Reply:
But where do those markers come from?

Reply:
Assuming it comes from Google maps api, it seems Google maps charge for every single piece of information requested. So if we have to turn all that off, I would think there might be a way to not ask those informations in the first place, which would limit our  billing

Reply:
> Assuming it comes from Google maps api, it seems Google maps charge for every single piece of information requested. So if we have to turn all that off, I would think there might be a way to not ask those informations in the first place, which would limit our billing

Map charges are based on the number of loads; hiding or displaying marker information does not affect it.
https://mapsplatform.google.com/pricing/